robottelo/host_info.py

```diff
@@ -99,28 +99,52 @@ def get_repo_files(repo_path, extension='rpm', hostname=None):
     return sorted(repo_file for repo_file in result.stdout if repo_file)
 
 
-def get_repomd_revision(repo_path, hostname=None):
-    """Fetches a revision of repository.
+def get_repo_files_by_url(url, extension='rpm'):
+    """Returns a list of repo files (for example rpms) in a specific repository
+    published at some url.
 
-    :param str repo_path: unix path to the repo, e.g. '/var/lib/pulp/fooRepo'
-    :param str optional hostname: hostname or IP address of the remote host. If
-        ``None`` the hostname will be get from ``main.server.hostname`` config.
+    :param url: url where the repo or CV is published
+    :param extension: extension of searched files. Defaults to 'rpm'
+    :return:  list representing rpm package names
+    """
+    if not url.endswith('/'):
+        url += '/'
+    result = ssh.command(
```

Comment:
Why run a curl command on the satellite server, instead of using the `requests` library to make the GET request from the python process this is running under?

Reply:
yes, of course, palmface at myself